iree/modules/check/native_module.cc

```diff
@@ -216,12 +216,12 @@ class CheckModuleState final {
     auto* lhs = lhs_ref.get();
     auto* rhs = rhs_ref.get();
     size_t lhs_rank = iree_hal_buffer_view_shape_rank(lhs);
-    absl::InlinedVector<int32_t, 6> lhs_shape(lhs_rank);
+    std::vector<int32_t> lhs_shape(lhs_rank);
     IREE_RETURN_IF_ERROR(
         iree_hal_buffer_view_shape(lhs, lhs_rank, lhs_shape.data(), nullptr));
```

Comment:
Ahhhh something weird is happening here. (see failing test [logs](https://source.cloud.google.com/results/invocations/1f9fc746-225e-42cc-918e-52819ed99c67/targets/iree%2Fgcp_ubuntu%2Fbazel%2Flinux%2Fx86-swiftshader%2Fcore%2Fpresubmit/log))

https://en.cppreference.com/w/cpp/container/vector/data
> If size() is 0, data() may or may not return a null pointer.

`lhs_rank` is actually `0`, so the vector size/capacity is `0` and `lhs_shape.data(0` returns `nullptr`

that's getting passed in here: https://github.com/google/iree/blob/1386d2cd284bf8babafdf90a0823afabd0e0c9f8/iree/hal/buffer_view.c#L161-L183

where `IREE_ASSERT_ARGUMENT(out_shape);` complains